erpnext/setup/doctype/customer_group/customer_group.js

```diff
@@ -1,37 +1,29 @@
-// Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
-// License: GNU General Public License v3. See license.txt
+// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
 
-
-cur_frm.cscript.refresh = function(doc, cdt, cdn) {
-	cur_frm.cscript.set_root_readonly(doc);
-}
-
-cur_frm.cscript.set_root_readonly = function(doc) {
-	// read-only for root customer group
-	if(!doc.parent_customer_group) {
-		cur_frm.set_read_only();
-		cur_frm.set_intro(__("This is a root customer group and cannot be edited."));
-	} else {
-		cur_frm.set_intro(null);
-	}
-}
-
-//get query select Customer Group
-cur_frm.fields_dict['parent_customer_group'].get_query = function(doc,cdt,cdn) {
-	return {
-		filters: {
-			'is_group': 1
+frappe.ui.form.on('Customer Group', {
+	setup: function(frm) {
+		frm.fields_dict['parent_customer_group'].get_query = function(doc) {
+			return {
+				filters:[
+					['Customer Group', 'is_group', '=', 1],
+					['Customer group', 'name', "!=", doc.customer_group_name]
+				]
+			}
```

Comment:
```suggestion
		frm.set_query('parent_customer_group', function(doc) {
			return {
				filters: {
					'is_group': 1,
					'name': ["!=", doc.customer_group_name]
				}
			}
		});
```

Do these changes for every query on the link field